pkg/clients/aws.go

```diff
@@ -80,14 +80,10 @@ const (
 // GetConfig constructs an *aws.Config that can be used to authenticate to AWS
 // API by the AWS clients.
 func GetConfig(ctx context.Context, c client.Client, mg resource.Managed, region string) (*aws.Config, error) {
-	switch {
-	case mg.GetProviderConfigReference() != nil:
+	if mg.GetProviderConfigReference() != nil {
 		return UseProviderConfig(ctx, c, mg, region)
-	case mg.GetProviderReference() != nil:
-		return UseProvider(ctx, c, mg, region)
-	default:
-		return nil, errors.New("neither providerConfigRef nor providerRef is given")
 	}
+	return nil, errors.New("neither providerConfigRef is given")
```

Comment:
@haarchri if we drop the `Provider` type then we likely should be removing `providerRef` as well, which is a `crossplane-runtime` change. Have there been any discussions / consensus reached on doing so?

cc @muvaf @negz

Reply:
This function was deprecated so i decided to do so - tell me i we want to drop

Reply:
@haarchri we may be getting close to the point that we want to drop it, but that is technically a breaking change and we may need to bump API versions because of that -- it might be worth decoupling that change from this PR so that your progress is not impeded by that discussion 👍🏻

Reply:
decoupling done - i will start a e2e test in our environment and will ping you if we are ready